src/Controller/ModulesController.php

```diff
@@ -354,6 +354,15 @@ protected function saveRelated(array $relatedData, string $id): void
             $method = (string)Hash::get($rel, 'method');
             $relation = (string)Hash::get($rel, 'relation');
             $relatedIds = (array)Hash::get($rel, 'relatedIds');
+
+            foreach ($relatedIds as $key => &$data) {
+                if (!empty($data['attributes'])) {
+                    $object = $this->apiClient->save($data['type'], array_merge([
+                        'id' => $data['id'] ?? null,
+                    ], $data['attributes']));
+                    $data['id'] = $object['data']['id'];
+                }
+            }
```

Comment:
this saves all related objects. why? the save logic should be performed by following `$this->apiClient->{$method}($id, $this->objectType, $relation, $relatedIds)`

Reply:
@edoardocavazza

Reply:
@didoda Does the POST call to update relationships update entities if attributes are passed?

Reply:
It can be handy, actually: you can create or update related objects while updating a relation in one shot.
I would `unset($data['attributes'])` after save though.

Reply:
Fair enough. Maybe we could add proper test cases to `ModulesControllerTest::testSaveRelated`